dcpy/models/product/metadata.py

```diff
@@ -113,19 +122,43 @@ class OrgMetadata(SortedSerializedBase, extra="forbid"):
     root_path: Path
     template_vars: dict = Field(default_factory=dict)
     metadata: OrgMetadataFile
+    column_defaults: dict[tuple[str, str], DatasetColumn]
+
+    @classmethod
+    def get_yml(cls, path: Path):
```

Comment:
I think I'd probably move this out to the module, and I'd definitely make it throw an exception if the path doesn't exist. Seems like a potential for masked errors otherwise?

Reply:
In this case I made this just to not write the exact same codes for the string snippets/column defaults, though it obviously is a little out of place. And it should be valid to not have snippets defined. I'll add an underscore to the name for now if that sounds okay? Or could skip this helper and just have slightly repeated code

Reply:
In this case I made this just to not write the exact same codes for the string snippets/column defaults, though it obviously is a little out of place. And it should be valid to not have snippets defined. I'll add an underscore to the name for now if that sounds okay? Or could skip this helper and just have slightly repeated code

Reply:
for sure, should be fine to not have snippets defined. This is more of a nit about having simple helper functions do one thing only (in this case, trying to read a yaml file), and having them complain when called incorrectly, e.g. when called with a nonexistent path. So I'd generally prefer to see something like:
```
snippets = read_yaml(snippets_path) if snippets_path.exists() else {}
```

it reminds me of a line I wrote, which I changed because of codecov. It went something like this:
```
def my_str_fn(s: str):
    if not s:
        return None
    else return s.some_str_fun(s)
```
Which I changed because it's up to the caller of `my_str_fn` to make sure they're passing in a valid string. (Well, I actually changed it because I was too lazy to write the unit test for the None case.)

Reply:
Yeah you're right. I just got rid of it... While it might be nice to have a `safe_load_path` throughout our repo, I didn't feel like making it just for these two lines.